x-pack/plugins/security_solution/public/management/pages/trusted_apps/store/middleware.test.ts

```diff
@@ -22,6 +24,14 @@ import { PaginationInfo, TrustedAppsListPageState } from '../state';
 import { initialTrustedAppsPageState, trustedAppsPageReducer } from './reducer';
 import { createTrustedAppsPageMiddleware } from './middleware';
 
+const initialNow = 111111;
+const dateNowMock = jest.fn();
+dateNowMock.mockReturnValue(initialNow);
+
+Date.now = dateNowMock;
```

Comment:
in investigating another unrelated issue, I did [find this approach in the Jest docs](https://jestjs.io/docs/en/snapshot-testing), so I guess its ok 🤷‍♂️

Reply:
Normally I prefer injections to using globals, which is a standard practice in Java. I would have an interface Clock or something like that and pass implementation of it to components that need it. Then it becomes pretty straight forward to test it. It is pretty cumbersome to do this with our style of using functions everywhere (as opposed to objects and interfaces). Hence I believe it would be perceived as over engineering. So given the usage of globals I'm not aware of better way of testing this in Jest.